### PR TITLE
Bugfix/ATG-123 Long press

### PR DESCRIPTION
**Pull request by IraCherginyshka**
Create LongPress directive to improve the game flow

### CODE REVIEW

File: src/app/directives/long-press.directive.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    if (this.moveStep !== FiguresMovement.ROTATE) {
+      this.interval = window.setInterval(() => {
+        this.gameService.setMoveStep(this.moveStep);
+      }, 125);
+    } else {
+      this.interval = window.setInterval(() => {
+        this.gameService.setMoveStep(FiguresMovement.ROTATE);
+      }, 250);
+    }
```
Comment (oleksandrstarov) [inline src/app/directives/long-press.directive.ts]: Please swap condition to be if (this.moveStep === FiguresMovement.ROTATE) {///} else  {} - such conditions is easier to read

File: src/app/directives/long-press.directive.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+export class LongPressDirective {
+  @Input('move') moveStep: FiguresMovement;
+
+  private interval: number;
+
+  constructor(private gameService: GameService) {}
+
+  @HostListener('touchstart', ['$event'])
+  @HostListener('mousedown', ['$event'])
+  onMouseDown(event: Event): void {
```
Comment (oleksandrstarov) [inline src/app/directives/long-press.directive.ts]: onMouseDown and endPress functions should be the same naming style I guess

**Review by oleksandrstarov — CHANGES_REQUESTED** [review]

File: src/app/directives/long-press.directive.ts (hunk 1 above)
Comment (IraCherginyshka) [reply src/app/directives/long-press.directive.ts]: Done

File: src/app/directives/long-press.directive.ts (hunk 2 above)
Comment (IraCherginyshka) [reply src/app/directives/long-press.directive.ts]: Changed

Comment (oleksandrstarov) [reply src/app/directives/long-press.directive.ts]: I am very grateful for your contribution, but please push your changes

**Review by oleksandrstarov — CHANGES_REQUESTED** [review]

File: src/app/directives/long-press.directive.ts (hunk 2 above)
Comment (IraCherginyshka) [reply src/app/directives/long-press.directive.ts]: Sorry, I have some problems with the code. Give me 10 minutes.

**Review by oleksandrstarov — APPROVED** [review]

**Pull request merged**